pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/results/ResultsBlockUtils.java

```diff
@@ -93,9 +99,17 @@ private static GroupByResultsBlock buildEmptyGroupByQueryResults(QueryContext qu
       columnDataTypes[index] = ColumnDataType.STRING;
       index++;
     }
-    for (AggregationFunction aggregationFunction : aggregationFunctions) {
+    for (int i = 0; i < aggregationFunctions.length; i++) {
       // NOTE: Use AggregationFunction.getResultColumnName() for SQL format response
-      columnNames[index] = aggregationFunction.getResultColumnName();
+      AggregationFunction aggregationFunction = aggregationFunctions[i];
+      String columnName = aggregationFunction.getResultColumnName();
+      if (filteredAggregationFunctions != null) {
```

Comment:
`filteredAggregationFunctions` should never be `null`. We can actually remove `aggregationFunctions` and always get `aggregationFunction` from `filteredAggregationFunctions`